pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -175,3 +199,13 @@ func (pc *PodCache) getPod(name string, namespace string) *v1.Pod {
 	}
 	return pod
 }
+
+func (pc *PodCache) dropNeedsUpdate(key string, ip string) {
```

Comment:
Trying to understand the what the function name implies.

The current logic is removing entry from the needResync map, but it does not trigger any update. If my understanding is correct, could the function name be like dropIPFromSyncQueue or similar?